rb/spec/integration/selenium/webdriver/driver_spec.rb

```diff
@@ -299,6 +299,22 @@ def save_screenshot_and_assert(path)
           end.to raise_error(Selenium::WebDriver::Error::ScriptTimeoutError)
         end
       end
+
+      describe 'window' do
+        types = ['tab', 'window'] # Stores the valid type values for new_window
+        types.each do |type|
+          it "should be able to open a new #{type}", only: {browser: %i[safari_preview]} do
+            # Store the count of window handles before executing action
+            before_window_handles = driver.window_handles.length
+            # Open a new window
+            driver.new_window(type)
+            # Store the count of window handles after executing action
+            after_window_handles = driver.window_handles.length
+            # 1 window handle should have been added
+            expect(after_window_handles).to eq(before_window_handles + 1)
```

Comment:
This test should probably also validate the type (If we're returning a full hash)

Reply:
Validating the type isn't really possible since it is driver implementation defined according to the webdriver spec. ie - just because :tab is passed in does not mean it will return :tab.

Reply:
Do we "know" what the current plans are, if so could we partition for that. Or is that too complex a question?

Reply:
@luke-hill We have no way of knowing what any specific webdriver compatible product may return, and limiting the test for no real benefit (especially since the plan is to ignore the returned type currently) doesn't seem to make sense.